File: multipicker/src/main/java/com/edwardstock/multipicker/picker/adapters/MediaFileDetailsLookup.java
package com.edwardstock.multipicker.picker.adapters;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.recyclerview.widget.RecyclerView;
import android.view.MotionEvent;
import android.view.View;

import com.edwardstock.multipicker.data.MediaFile;

import java.lang.ref.WeakReference;
import androidx.recyclerview.selection.ItemDetailsLookup;

/**
 * android-image-picker. 2018
 * @author Eduard Maximovich [edward.vstock@gmail.com]
 */
public class MediaFileDetailsLookup extends ItemDetailsLookup<MediaFile> {
    private WeakReference<RecyclerView> mRecyclerView;

    public MediaFileDetailsLookup(RecyclerView recyclerView) {
        mRecyclerView = new WeakReference<>(recyclerView);
    }

    @Nullable
    @Override
    public ItemDetails<MediaFile> getItemDetails(@NonNull MotionEvent e) {
        View view = mRecyclerView.get().findChildViewUnder(e.getX(), e.getY());
        if (view != null) {
            RecyclerView.ViewHolder viewHolder = mRecyclerView.get().getChildViewHolder(view);
            if (viewHolder instanceof FilesAdapter.ViewHolderWithDetails) {
                int position = viewHolder.getAdapterPosition();
                return ((FilesAdapter.ViewHolderWithDetails<MediaFile>) viewHolder).getItemDetails(((FilesAdapter) mRecyclerView.get().getAdapter()), position);
            }
        }

        return null;

    }
}

File: multipicker/src/main/java/com/edwardstock/multipicker/picker/adapters/MediaFileDetailsLookup.kt
package com.edwardstock.multipicker.picker.adapters

import android.view.MotionEvent
import androidx.recyclerview.selection.ItemDetailsLookup
import androidx.recyclerview.widget.RecyclerView
import com.edwardstock.multipicker.data.MediaFile
import com.edwardstock.multipicker.picker.adapters.FilesAdapter.ViewHolderWithDetails
import java.lang.ref.WeakReference

/**
 * android-image-picker. 2018
 * @author Eduard Maximovich [edward.vstock@gmail.com]
 */
class MediaFileDetailsLookup(recyclerView: RecyclerView) : ItemDetailsLookup<MediaFile>() {

    private val mRecyclerView: WeakReference<RecyclerView> = WeakReference(recyclerView)

    override fun getItemDetails(e: MotionEvent): ItemDetails<MediaFile>? {
        val view = mRecyclerView.get()!!.findChildViewUnder(e.x, e.y)
        if (view != null) {
            val viewHolder = mRecyclerView.get()!!.getChildViewHolder(view)
            if (viewHolder is ViewHolderWithDetails<*>) {
                val position = viewHolder.bindingAdapterPosition
                return (viewHolder as ViewHolderWithDetails<MediaFile>).getItemDetails(mRecyclerView.get()!!.adapter!! as FilesAdapter, position)
            }
        }
        return null
    }

}